examples/ag-ui/servers/php/lib/Providers.php

```
@@ -0,0 +1,190 @@
+
+    $ch = curl_init($url);
+    if ($ch === false) {
+        throw new RuntimeException('failed to initialize curl');
+    }
+
+    curl_setopt_array($ch, [
+        CURLOPT_POST => true,
+        CURLOPT_HTTPHEADER => $headers,
+        CURLOPT_POSTFIELDS => $payload,
+        CURLOPT_RETURNTRANSFER => false,
+        CURLOPT_WRITEFUNCTION => static function ($handle, string $chunk) use (&$buffer, $handleEvent): int {
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$handle'. (undefined)

(UnusedFormalParameter)